types/params_test.go

```diff
@@ -23,23 +23,140 @@ func TestConsensusParamsValidation(t *testing.T) {
-		7: {makeParams(1024*1024*1024, 0, -1, 0, valEd25519), false},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   1,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: true,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   0,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: false,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   47 * 1024 * 1024,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: true,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   10,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: true,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   100 * 1024 * 1024,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: true,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   101 * 1024 * 1024,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: false,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   1024 * 1024 * 1024,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: false,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:   1024 * 1024 * 1024,
+				evidenceAge:  2,
+				precision:    1,
+				messageDelay: 1}),
+			valid: false,
+		},
 		// test evidence params
-		8:  {makeParams(1, 0, 0, 0, valEd25519), false},
-		9:  {makeParams(1, 0, 2, 2, valEd25519), false},
-		10: {makeParams(1000, 0, 2, 1, valEd25519), true},
-		11: {makeParams(1, 0, -1, 0, valEd25519), false},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:       1,
+				evidenceAge:      0,
+				maxEvidenceBytes: 0,
+				precision:        1,
+				messageDelay:     1}),
+			valid: false,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:       1,
+				evidenceAge:      2,
+				maxEvidenceBytes: 2,
+				precision:        1,
+				messageDelay:     1}),
+			valid: false,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:       1000,
+				evidenceAge:      2,
+				maxEvidenceBytes: 1,
+				precision:        1,
+				messageDelay:     1}),
+			valid: true,
+		},
+		{
+			params: makeParams(makeParamsArgs{
+				blockBytes:       1,
+				evidenceAge:      -1,
+				maxEvidenceBytes: 0,
+				precision:        1,
+				messageDelay:     1}),
+			valid: false,
+		},
 		// test no pubkey type provided
-		12: {makeParams(1, 0, 2, 0, []string{}), false},
+		{
+			params: makeParams(makeParamsArgs{
+				evidenceAge:  2,
+				pubkeyTypes:  []string{},
+				precision:    1,
+				messageDelay: 1}),
+			valid: false,
+		},
 		// test invalid pubkey type provided
-		13: {makeParams(1, 0, 2, 0, []string{"potatoes make good pubkeys"}), false},
+		{
+			params: makeParams(makeParamsArgs{
+				evidenceAge:  2,
+				pubkeyTypes:  []string{"potatoes make good pubkeys"},
```

Comment:
😂

öffentlicher Kartoffelschlüssel

Reply:
?????

Reply:
I was just amused by the notion of a public potato key.